drivers/media/platform/raspberrypi/rp1_cfe/cfe.c

```diff
@@ -1826,7 +1826,7 @@ static void cfe_unregister_nodes(struct cfe_device *cfe)
 
 static int cfe_link_node_pads(struct cfe_device *cfe)
 {
-	unsigned int i;
+	unsigned int i, source_pad = 0;
```

Comment:
But the logic depends on it not being reinitialised between iterations of the outer loop.

Reply:
It would be in the outer for loop, ie
```
for (i = 0, source_pad = 0; i < CSI2_NUM_CHANNELS; i++) {
```
which also looks a little strange.

Quite happy to swap to a while loop.

Reply:
Done